components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -176,17 +168,17 @@ bool BraveReferralsHeaders::GetMatchingReferralHeaders(
       LOG(WARNING) << "Failed to retrieve 'headers' key from referral headers";
       continue;
     }
+
+    url_matcher::URLMatcherConditionFactory condition_factory;
     for (const auto& domain_value : *domains_list) {
-      URLPattern url_pattern(URLPattern::SCHEME_HTTPS |
-                             URLPattern::SCHEME_HTTP);
-      url_pattern.SetScheme("*");
-      url_pattern.SetHost(domain_value.GetString());
-      url_pattern.SetPath("/*");
-      url_pattern.SetMatchSubdomains(true);
-      if (!url_pattern.MatchesURL(url))
-        continue;
-      *request_headers_dict = headers_dict;
-      return true;
+      if (auto url_match_conditions =
+              condition_factory.CreateHostSuffixCondition(
+                  domain_value.GetString());
+          url_match_conditions.IsMatch(
+              {0 /* default ID of condition set by condition factory*/}, url)) {
+        *request_headers_dict = headers_dict;
+        return true;
+      }
```

Comment:
I'm not sure about this part as it's my first time to use matcher. are there any tests covering this change?

Reply:
BraveReferralsNetworkDelegateHelperTest.ReplaceHeadersForMatchingDomain looks convering.